graylog2-web-interface/src/components/configurations/EventsConfig.jsx

```diff
@@ -0,0 +1,183 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import createReactClass from 'create-react-class';
+import { Button, FormGroup, HelpBlock } from 'react-bootstrap';
+import { BootstrapModalForm } from 'components/bootstrap';
+import { IfPermitted, TimeUnitInput } from 'components/common';
+import ObjectUtils from 'util/ObjectUtils';
+import FormUtils from 'util/FormsUtils';
+import Input from 'components/bootstrap/Input';
+import { extractDurationAndUnit } from 'components/common/TimeUnitInput';
+import lodash from 'lodash';
+import moment from 'moment';
+
+const TIME_UNITS = ['HOURS', 'MINUTES', 'SECONDS'];
+
+const EventsConfig = createReactClass({
+  displayName: 'EventsConfig',
+
+  propTypes: {
+    config: PropTypes.shape({
+      events_search_timeout: PropTypes.number,
+      events_notification_retry_period: PropTypes.number,
+      events_notification_default_backlog: PropTypes.number,
+    }),
+    updateConfig: PropTypes.func.isRequired,
+  },
+
+  getDefaultProps() {
+    return {
+      config: {
+        events_search_timeout: 60000,
+        events_notification_retry_period: 300000,
+        events_notification_default_backlog: 50,
+      },
+    };
+  },
+
+  getInitialState() {
+    const { config } = this.props;
+    return {
+      config: ObjectUtils.clone(config),
```

Comment:
In this case you are not modifying the `config` prop, so you don't need to clone it.